src/storage/user-data.ts

```diff
@@ -86,7 +87,17 @@ export async function saveUserData( data: UserData ): Promise< void > {
 	const filePath = getUserDataFilePath();
 
 	const asString = JSON.stringify( toDiskFormat( data ), null, 2 ) + '\n';
-	await fs.promises.writeFile( filePath, asString, 'utf-8' );
+	try {
+		await atomically.writeFile( filePath, asString, 'utf-8' );
+	} catch ( error ) {
+		// Fall back to FS function in case the writing fails with EXDEV error.
+		// This issue might happen on Windows when renaming a file.
+		// Reference: https://github.com/sindresorhus/electron-store/issues/106
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		if ( ( error as any )?.code === 'EXDEV' ) {
+			await fs.promises.writeFile( filePath, asString, 'utf-8' );
+		}
+	}
```

Comment:
This implementation is based on [`conf`](https://github.com/sindresorhus/conf) package (used by [`electron-store`](https://github.com/sindresorhus/electron-store)):
https://github.com/sindresorhus/conf/blob/608adb0c46fb1680ddbd9833043478367a64c120/source/index.ts#L456-L467